### PR TITLE
Remove SessionManager from SitePermissionsFeature.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt [hunk 1]
```diff
@@ -88,8 +87,7 @@ class SitePermissionsFeature(
     private val dialogConfig: DialogConfig? = null,
     override val onNeedToRequestPermissions: OnNeedToRequestPermissions,
     val onShouldShowRequestPermissionRationale: (permission: String) -> Boolean,
-    private val store: BrowserStore,
-    private val customTabId: String? = null
```
Comment (Amejia481) [inline components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt:92]: In this case we have `sessionId` and `customTabId` doing the same behaviour.

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt [hunk 2]
```diff
@@ -682,12 +682,8 @@ class SitePermissionsFeature(
      * Re-attaches a fragment that is still visible but not linked to this feature anymore.
      */
     private fun reattachFragment(fragment: SitePermissionsDialogFragment) {
-        val session = sessionManager.findSessionById(fragment.sessionId)
-        val contentState = getCurrentContentState()
-
-        if (session == null || contentState == null ||
-            (noPermissionRequests(contentState))
-        ) {
+        val currentTab = store.state.findTabOrCustomTabOrSelectedTab(fragment.sessionId)?.content
```
Comment (Amejia481) [inline components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt:685]: Here we need to warranty that when the provided `sessionId` is not found we have to fallback to selected tab.

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt [hunk 3]
```diff
@@ -172,7 +170,7 @@ class SitePermissionsFeature(
         appPermissionRequest: PermissionRequest,
         optionalSessionId: String? = null
     ) {
-        val thisSessionId = optionalSessionId ?: sessionId
+        val thisSessionId = optionalSessionId ?: getCurrentTabState()?.id
```
Comment (Amejia481) [inline components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt:173]: Here we need to warranty that when the provided `sessionId` is not found we have to fallback to selected tab.

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt [hunk 4]
```diff
@@ -161,7 +159,7 @@ class SitePermissionsFeature(
         permissionRequest: PermissionRequest,
         optionalSessionId: String? = null
     ) {
-        val thisSessionId = optionalSessionId ?: sessionId
+        val thisSessionId = optionalSessionId ?: getCurrentTabState()?.id
```
Comment (Amejia481) [inline components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt:162]: Here we need to warranty that when the provided `sessionId` is not found we have to fallback to selected tab.

**Review by codrut-topliceanu — APPROVED** [review]
Looks great, works great. Awesome! 👍

**Pull request merged**